apps/editing-toolkit/editing-toolkit-plugin/error-reporting/index.php

```diff
@@ -40,45 +48,94 @@ function add_crossorigin_to_script_els( $tag ) {
 	return $tag;
 }
 
+/**
+ * Temporary function to feature flag Sentry by segment. We'll be testing
+ * it on production (simple sites) for a while to see if it's feasible to
+ * activate it for all sites and perhaps get rid of our custom solution.
+ * If it works well, we'll activate for all simple sites and look into.
+ * activating it for WoA, too.
+ *
+ * @param int $user_id the user id.
+ * @return bool
+ */
+function user_in_sentry_test_segment( $user_id ) {
+	$current_segment = 1; // segment of existing users that will get this feature in %.
+	$user_segment    = $user_id % 100;
+
+	// We get the last two digits of the user id and that will be used to decide in what
+	// segment the user is. i.e if current_segment is 10, then only ids that end in < 10
+	// will be considered part of the segment.
+	return $user_segment < $current_segment;
+}
+
+/**
+ * Returns whether or not the site loading ETK is in the WoA env.
+ *
+ * @return bool
+ */
+function is_atomic() {
+	return defined( 'IS_ATOMIC' ) && IS_ATOMIC;
+}
+/**
+ * Return whether Sentry should be activated for a given user.
+ *
+ * In this phase, a12s have the possibility of configuring what error reporter to choose
+ * through the sticker. a12s shouldn not be covered by the segment logic.
+ *
+ * Regular users have the error reporter chosen based on the segmentation logic, only.
+ *
+ * @param int $user_id The user id. Used to check if the user is A8C or in
+ * the Sentry test segment.
+ * @param int $blog_id The blog ID. Usually the value of `get_current_blog_id`.
+ * Used to check if the sticker is applied if user is A8C.
+ */
+function should_activate_sentry( $user_id, $blog_id ) {
+	return ( is_automattician( $user_id ) && has_blog_sticker( 'error-reporting-use-sentry', $blog_id ) )
+		|| ( ! is_automattician( $user_id ) && user_in_sentry_test_segment( $user_id ) );
```

Comment:
Does it make sense to just have it be `has_blog_sticker || user_in_sentry_test_segment`?

Reply:
🤔 I guess we could, but there would be a very small probability of an a11n falling into the `user_sentry_test_segment` if it doesn't have the sticker and forcedly have Sentry activated even when semantically it should have the "classic/homebrew" error reporting.

Reply:
To add to that: I don't simplifying the expression as you suggest would be bad and that the possible consequences are a big deal, but I also think it's fine to leave as is since this logic is ephemerous: it'll be removed as soon as we decide what path to go: Sentry or Classic (or both).

Reply:
Sounds good!